### PR TITLE
document the set-data and append-data commands in patch

### PR DESCRIPTION
**Pull request by whyrusleeping**
achin in irc mentioned that he wish he could do 'ipfs object patch $hash set-data < file.dat'.

Its totally a thing already, so heres some docs.

License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

**Comment by eminence** [comment]
:smile:

**Comment by whyrusleeping** [comment]
@jbenet ping!

File: core/commands/object.go [hunk 1]
```diff
@@ -450,6 +452,12 @@ a file containing 'bar', and returns the hash of the new object.
 
 This removes the link named foo from the hash in $FOO_BAR and returns the
 resulting object hash.
+
+The data inside the node can be modified as well:
+
+    ipfs object patch $FOO_BAR set-data < file.dat
+	ipfs object patch $FOO_BAR append-data < file.dat
```
Comment (jbenet) [inline core/commands/object.go]: indent fix

File: core/commands/object.go [hunk 2]
```diff
@@ -433,9 +433,11 @@ var objectPatchCmd = &cmds.Command{
 	Helptext: cmds.HelpText{
 		Tagline: "Create a new merkledag object based on an existing one",
 		ShortDescription: `
-'ipfs object patch <root> [add-link|rm-link] <args>' is a plumbing command used to
+'ipfs object patch <root> <cmd> <args>' is a plumbing command used to
```
Comment (jbenet) [inline core/commands/object.go]: need to document all the commands, like:

```
patch commands

    add-link <name> <value>    add a link to a node
    rm-link <name>              ...
    set-data                          ... mention stdin
    append-data                   ... mention stdin
```

**Comment by whyrusleeping** [comment]
@jbenet comments addressed.

**Pull request merged**